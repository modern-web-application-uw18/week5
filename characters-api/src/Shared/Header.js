import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import  './shared.css';

class Header extends Component {
    render() {  
        return (
            <div className = 'header'>
            <h1> <Link to="/">Rick and Morty Character Almanac</Link> </h1>
            </div>
        )
    }
}        

export default Header;