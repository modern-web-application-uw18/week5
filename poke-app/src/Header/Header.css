header a {
  color: rgb(196, 196, 196);
  border: 1px solid rgb(196, 196, 196);
  padding: 5px;
  border-radius: 10px;
  text-decoration: none;
}
