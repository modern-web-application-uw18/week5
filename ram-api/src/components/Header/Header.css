.Header {
    height: 56px;
    width: 100%;
    position: fixed;
    top: 0;
    left: 0;
    background-color: #272727;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 0 20px;
    box-sizing: border-box;
    z-index: 90;
}

.Header nav {
    height: 100%;
}

.HeaderTitle {
    color: #4e4e4e;
    font-size: 22px;
    margin: 0px;
}


@media (max-width: 668px) {
    .HeaderTitle {
        font-size: 15px;
    }


}