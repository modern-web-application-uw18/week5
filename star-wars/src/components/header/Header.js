import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import './header.css';

class Header extends Component {

  render() {
    return (
      <header className='site-header'>
        <Link to="/">Characters</Link>
        <Link to="/films">Films</Link>
      </header>
    );
  }
}

export default Header;
