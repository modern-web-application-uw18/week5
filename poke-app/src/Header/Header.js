import React, { Component } from 'react';
import logo from './pokeball.png';

import './Header.css';

export default class Header extends Component {

    render() {
        return (
          <header className="App-header">
            <img src={logo} className="App-logo" alt="logo" />
            <h1 className="App-title">Welcome to my Pokedex</h1>
          </header>
        );
    }
}
