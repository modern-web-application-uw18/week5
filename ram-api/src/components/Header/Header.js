import React, { Component } from 'react';
import './Header.css';
import NavigationItems from './NavigationItems/NavigationItems';



class Header extends Component {
    render() {
        return (
            <header className="Header">
                <h1 style={{color:"gray"}}>R&M API</h1>
                <nav>
                    <NavigationItems />
                </nav>
            </header>
        )
    }
}

export default Header;
