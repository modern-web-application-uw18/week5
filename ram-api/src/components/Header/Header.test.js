import React from 'react';
import { mount } from 'enzyme';
import Header from './Header'
import { MemoryRouter} from 'react-router-dom';
// import characters from '../../__mocks__/characters.json'

describe('<Header />', () => {
    it('Should render Header', () => {

        const wrapper = mount(
            <MemoryRouter> 
                <Header searchHandler={jest.fn()} />
            </MemoryRouter>
        );


        expect(wrapper).toMatchSnapshot();
    })
})




