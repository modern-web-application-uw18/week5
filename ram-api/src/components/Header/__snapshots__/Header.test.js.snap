// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should render Header 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "jhy6uj",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "jhy6uj",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(Header)
      searchHandler={[Function]}
    >
      <Route
        render={[Function]}
      >
        <Header
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "canGo": [Function],
              "createHref": [Function],
              "entries": Array [
                Object {
                  "hash": "",
                  "key": "jhy6uj",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
              ],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "index": 0,
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "key": "jhy6uj",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "key": "jhy6uj",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
          searchHandler={[Function]}
        >
          <header
            className="Header"
          >
            <SearchBox
              searchHandler={[Function]}
            >
              <div
                className="SearchBox"
              >
                <div
                  className="row"
                >
                  <div
                    className="col-md-12"
                  >
                    <div
                      className="input-group"
                    >
                      <input
                        className="form-control"
                        onChange={[Function]}
                        placeholder="Filter by character name..."
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </SearchBox>
            <span
              className="HeaderTitle"
            >
              R&M Attack
            </span>
            <nav>
              <navigationItems>
                <ul
                  className="NavigationItems"
                >
                  <navigationItem
                    exact={true}
                    link="/"
                  >
                    <li
                      className="NavigationItem"
                    >
                      <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        exact={true}
                        to="/"
                      >
                        <Route
                          exact={true}
                          path="/"
                        >
                          <Link
                            aria-current="true"
                            className="active"
                            replace={false}
                            style={Object {}}
                            to="/"
                          >
                            <a
                              aria-current="true"
                              className="active"
                              href="/"
                              onClick={[Function]}
                              style={Object {}}
                            >
                              Home
                            </a>
                          </Link>
                        </Route>
                      </NavLink>
                    </li>
                  </navigationItem>
                  <navigationItem
                    link="/detail"
                  >
                    <li
                      className="NavigationItem"
                    >
                      <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/detail"
                      >
                        <Route
                          path="/detail"
                        >
                          <Link
                            aria-current={false}
                            replace={false}
                            to="/detail"
                          >
                            <a
                              aria-current={false}
                              href="/detail"
                              onClick={[Function]}
                            >
                              Detail
                            </a>
                          </Link>
                        </Route>
                      </NavLink>
                    </li>
                  </navigationItem>
                  <navigationItem
                    link="/favs"
                  >
                    <li
                      className="NavigationItem"
                    >
                      <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        to="/favs"
                      >
                        <Route
                          path="/favs"
                        >
                          <Link
                            aria-current={false}
                            replace={false}
                            to="/favs"
                          >
                            <a
                              aria-current={false}
                              href="/favs"
                              onClick={[Function]}
                            >
                              Favorites
                            </a>
                          </Link>
                        </Route>
                      </NavLink>
                    </li>
                  </navigationItem>
                </ul>
              </navigationItems>
            </nav>
          </header>
        </Header>
      </Route>
    </withRouter(Header)>
  </Router>
</MemoryRouter>
`;
